### DISCUSSION TITLE
Adding html-survey-form data using jsPsych.data.addProperties

### DISCUSSION THREAD

**Discussion by amunn** [jspsych/jsPsych#2723] [Q&A] [answered]
We collect demographic data using the `html-survey-form` plugin. This generates a single `response` with keys for each of the form items. Right now, we unpack this in R after the data is collected, but it seems to be that it would more efficient to add each of the form item to the final experiment data within the experiment using `jsPsych.data.addProperties`, but I'm not quite sure how to do that.

e.g.  from the following form, I would like to add `test_date` and `experimenter_name` to each item of the final collected data.

```
var questions =
	'<div id="demographic_form" name="demo_form">'+
	'<ol>' +
	'<li><div id="demo_block"><div id="demo_question_block">Test Date <input name="test_date" type="date" /></li>' +
	'<li> Experimenter name<input name="experimenter_name" type="text" /></li>' +
	'</ol>'+
	'</div>';

var demographic_trial = {
	type: jsPsychSurveyHtmlForm,
	html: questions
	};
```

**Comment by amunn** [comment, marked as answer]
Adapting the method found in #1699 it seems that I can do the following:

```
var jsPsych = initJsPsych({
   on_finish: function() {
       demo_data = jsPsych.data.get().filter({trial_type: 'survey-html-form'}).values()[0].response;
       jsPsych.data.addProperties({
            test_date: demo_data.test_date,
            experimenter_name: demo_data.experimenter_name
       });
     jsPsych.data.displayData();
}});
```

**Reply by becky-gilbert** [reply]
Yes this is what would suggest. Glad you figured it out, and thanks for the update!
